tests/functional/helpers.py

```diff
@@ -40,15 +36,37 @@ def safe_delete(
         except gitlab.exceptions.GitlabGetError:
             return
 
+        logging.info(f"Deleting object type {type(object)}")
+
         if index:
             logging.info(f"Attempt {index + 1} to delete {object!r}.")
         try:
-            if hard_delete:
+            if type(object) is gitlab.v4.objects.User:
+                # You can't use this option if the selected user is the sole owner of any groups
+                # Use `hard_delete=True` or a 'Ghost User' may be created.
+                # https://docs.gitlab.com/ee/api/users.html#user-deletion
                 object.delete(hard_delete=True)
+                if index > 1:
+                    # If User is the sole owner of any group it won't be deleted,
+                    # which combined with parents group never immediately deleting in GL 16
+                    # we shouldn't cause test to fail if it still exists
+                    return
+            elif type(object) is gitlab.v4.objects.Project:
```

Comment:
nit: Any reason not to use `if isinstance(object, gitlab.v4.objects.Project)` ?

Reply:
nope - my lack of embedded python knowledge when I first did the MR 👍